index.bs

```diff
@@ -306,14 +306,38 @@ partial interface XRSession {
   readonly attribute XRDepthUsage depthUsage;
   readonly attribute XRDepthDataFormat depthDataFormat;
   readonly attribute XRDepthType? depthType;
+  readonly attribute boolean? depthActive;
+
+  undefined startDepth();
+  undefined stopDepth();
 };
 </script>
 
 The {{XRSession/depthUsage}} describes depth sensing usage with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}.
 
 The {{XRSession/depthDataFormat}} describes depth sensing data format with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}.
 
-The {{XRSession/depthType}} describes the depth sensing type with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}. If the runtime only supports a single {{XRDepthType}} or otherwise ignored {XRDepthStateInit/depthTypeRequest}} this may return <code>null</code>.
+The {{XRSession/depthType}} describes the depth sensing type with which the session was configured. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}. If the runtime only supports a single {{XRDepthType}} or otherwise ignored {{XRDepthStateInit/depthTypeRequest}} this may return <code>null</code>.
+
+The {{XRSession/depthActive}} returns the current [=depth sensing active state=]. If this attribute is accessed on a session that does not have depth sensing enabled, the user agent MUST throw an {{InvalidStateError}}.  When this value is <code>false</code>, the user agent MUST reject attempts to get depth data. When this value is <code>true</code> the user agent MAY return depth data that is <code>null</code>.
+
+<div class="algorithm" data-algorithm="start-depth-sensing">
+
+When {{XRSession/startDepth()}} is invoked on an {{XRSession}} |session|, the User Agent MUST run the following steps:
+
+1. If |session|'s {{XRSession/ended}} value is <code>true</code>, throw an {{InvalidStateError}} and abort these steps.
+1. If [=depth-sensing=] feature descriptor is not [=list/contain|contained=] in the |session|'s [=XRSession/XR device=]'s [=XR device/list of enabled features=] for |session|'s [=XRSession/mode=], [=exception/throw=] a {{NotSupportedError}} and abort these steps.
+1. If [=depth sensing active state=] is <code>true</code>, throw an {{InvalidStateError}} and abort these steps.
```

Comment:
Feels a bit strange to have a method that can only be safely called if you check an attribute each time, and which only serves to change the state of said attribute. At that point why not just set the attribute directly? (Yes, yes, I know that was already discussed...)

If these are going to be separate methods, then it seems like calling `start` after depth sensing is already started is pretty harmless and probably shouldn't throw. Same for `stop`.

Reply:
Ack.